web/src/engine/keyboard/src/keyboards/keyboardLoaderBase.ts

```diff
@@ -76,23 +16,49 @@ export default abstract class KeyboardLoaderBase {
     this._harness = harness;
   }
 
-  public loadKeyboardFromPath(uri: string): Promise<Keyboard> {
+  /**
+   * Load a keyboard from a remote or local URI.
+   *
+   * @param uri  The URI of the keyboard to load.
+   * @returns    A Promise that resolves to the loaded keyboard.
+   */
+  public async loadKeyboardFromPath(uri: string): Promise<Keyboard> {
```

Comment:
Technically we only need the `async` keyword if we use `await` or to automatically wrap return values in a `Promise<>`. Given we are doing neither here, we could drop this.

```suggestion
  public loadKeyboardFromPath(uri: string): Promise<Keyboard> {
```